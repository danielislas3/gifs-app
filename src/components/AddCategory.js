import React from "react";
import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {


  const [inputValue, setInputValue] = useState("Hola mundo");
  
  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = e => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cat) => [...cat, inputValue]);
      setInputValue('')
    }

    console.log(e);
  };
  
  return (
    <form onSubmit={handleSubmit}>
      <h2>Add Category</h2>
      <input
        name="cat"
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes={
  setCategories:PropTypes.func.isRequired
}