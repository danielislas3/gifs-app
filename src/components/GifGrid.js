import React from "react";
import PropTypes from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data:gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid animate__animated animate__fadeIn">
        {gifs.map(gif => (
          <GifGridItem {...gif} key={gif.id} />
        ))}
      </div>
    </>
  );
};
GifGrid.propTypes={
  category:PropTypes.string.isRequired
}
