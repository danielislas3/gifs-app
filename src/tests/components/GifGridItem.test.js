import '@testing-library/jest-dom'
import React from 'react'
import {shallow} from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas en <GifGridItem></GifGridItem>', () => {
  const title ='Un titulo'
  const url ='http//'
  test('Debe mostrar el copmponenete correctamnte', () => {
    
const wrapper = shallow(<GifGridItem title={title} url={url}/>) 

   expect(wrapper).toMatchSnapshot()
  })
  
})
