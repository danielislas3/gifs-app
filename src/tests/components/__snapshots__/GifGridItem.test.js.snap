// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem></GifGridItem> Debe mostrar el copmponenete correctamnte 1`] = `
<div
  className="card"
>
  <img
    alt="Un titulo"
    src="http//"
  />
  <p>
    Un titulo
  </p>
</div>
`;
